src/QirRuntime/samples/StandaloneInputReference/qir-driver.cpp

```diff
@@ -0,0 +1,141 @@
+
+    // Add the options that correspond to the parameters that the QIR entry-point needs.
+    // Option for a Q# Int type.
+    int64_t intValue = 0;
+    app.add_option("--int-value", intValue, "An integer value")->required();
+
+    // Option for a Q# Double type.
+    double_t doubleValue = 0.0;
+    app.add_option("--double-value", doubleValue, "A double value")->required();
+
+    // Option for a Q# Bool type.
+    bool boolValue = false;
+    app.add_option("--bool-value", boolValue, "A bool value")->required();
+
+    // Option for Q# Pauli type.
+    PauliId pauliValue = PauliId::PauliId_I;
+    app.add_option("--pauli-value", pauliValue, "A Pauli value")
+        ->required()
+        ->transform(CLI::CheckedTransformer(PauliMap, CLI::ignore_case));
+
+    // Option for Q# Range type.
+    tuple<int64_t, int64_t, int64_t> rangeValue(0, 0, 0);
+    app.add_option("--range-value", rangeValue, "A Range value (start, step, end)")->required();
+    QirRange qirRange = {
+        get<0>(rangeValue), // Start
+        get<1>(rangeValue), // Step
+        get<2>(rangeValue)  // End
+    };
+
+    // Option for Q# Result type.
+    bool resultValue = false;
+    app.add_option("--result-value", resultValue, "A Result value")
+        ->required()
+        ->transform(CLI::CheckedTransformer(ResultMap, CLI::ignore_case));
+
+    Result result = resultValue ? sim->UseOne() : sim->UseZero();
+
+    // Option for Q# String type.
+    string stringValue;
+    app.add_option("--string-value", stringValue, "A String value")->required();
+    QirString* qirString = quantum__rt__string_create(stringValue.c_str());
+
+    // Option for a Q# Array<Int> type.
+    vector<int64_t> integerArray;
+    CLI::Option* integerArrayOpt = app.add_option("--integer-array", integerArray, "An integer array")->required();
+
+    // With all the options added, parse arguments from the command line.
+    CLI11_PARSE(app, argc, argv);
+
+    // Redirect the simulator output from std::cout if the --simulator-output option is present.
+    ostream* simulatorOutputStream = &cout;
+    filebuf simulationOutputFileBuffer;
+    if (!simulationOutputFileOpt->empty())
+    {
+        simulationOutputFileBuffer.open(simulationOutputFile, ios::out);
+        ostream simulationOutputFileStream(&simulationOutputFileBuffer);
+        Quantum::Qis::Internal::SetOutputStream(simulationOutputFileStream);
+        simulatorOutputStream = &simulationOutputFileStream;
+    }
+
+    // Redirect the Q# operation output from std::cout if the --operation-output option is present.
+    ostream* operationOutputStream = &cout;
+    filebuf operationOutputFileBuffer;
+    if (!operationOutputFileOpt->empty())
+    {
+        operationOutputFileBuffer.open(operationOutputFile, ios::out);
+        ostream operationOutputFileStream(&operationOutputFileBuffer);
+        operationOutputStream = &operationOutputFileStream;
+    }
```

Comment:
I'm not strong at streams, but at a glance I have an impression that
* the `operationOutputFileStream` is a local variable on the stack that will survive until the nearest `}`
* the `operationOutputStream` starts pointing to that local variable on the stack (in line 122);
* upon `}` the `operationOutputFileStream` goes out of scope and gets destroyed, after which the `operationOutputStream` becomes a dangling pointer (pointing to the non-existent variable on the stack).

If I'm mistaken then at least a comment clarifying the situation, would help.

Reply:
Great catch! I changed it, and now it uses an `ofstream` that can be declared in the scope of the `main` function and opened inside the `if`.